src/php/Compiler/Analyzer/TypeAnalyzer/SpecialForm/FnSymbol.php

```diff
@@ -63,21 +65,21 @@ private function analyzeBody(FnSymbolTuple $fnSymbolTuple, RecurFrame $recurFram
         return $this->analyzer->analyze($body, $bodyEnv);
     }
 
-    private function createDoTupleWithBody(array $body): Tuple
+    private function createDoTupleWithBody(array $body): PersistentListInterface
     {
-        return Tuple::create(
+        return TypeFactory::getInstance()->persistentListFromArray([
             (Symbol::create(Symbol::NAME_DO))->copyLocationFrom($body),
-            ...$body
-        )->copyLocationFrom($body);
+            ...$body,
```

Comment:
> ERROR: DuplicateArrayKey - src/php/Compiler/Analyzer/TypeAnalyzer/SpecialForm/FnSymbol.php:72:16 - String keys are not supported in unpacked arrays (see https://psalm.dev/151)
            ...$body,
            
Maybe you might fix these psalm errors by using the values from the array? 🤔 

```php
...array_values($body)
```

Reply:
No it is just enough to change type annotation from `MyType[]` to `array<int, MyType>`. This came up with Psalm 4.7. It fixed this already in master but was to tired to fix this here.

Reply:
I can help you this evening with this :)

Reply:
Fixed it.